### PR TITLE
Documentation Update

### PR DESCRIPTION
**Pull request by GammaSigma1234**
@dughiboogie 
@DanieleAsnicar 

Several improvements and additions to the Project's documentation. Still work in progress.

### CODE REVIEW

**Comment by GammaSigma1234** [comment]
Comment added to test the "resolve conversation" rule.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,63 @@
...
+- [Use Cases](#use-cases)
+- [Requirements](#requirements)
+- [Ideas for Improvement](#ideas-for-improvement)
+
+
+# Goal Of The Repository
+
+This repository describes the implementation of an open-source Memory game in Qt, applying principles of software architecture such as requirements gathering and object-oriented analysis and design.
+
+This project uses the [Qt Widgets? / Qt Quick? TBD] approach.
```
Comment (dughiboogie) [inline README.md]: After last meeting I'd say we can probably commit to Qt Quick?
```suggestion
This project uses the Qt Quick approach.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,63 @@
...
+
+[Use cases](#use-cases) and [requirements](#requirements) are kept in the documentation directory (`docs`).
+
+# Description Of The Application
+
+Memory (also known as Concentration) is a card game in which all cards are dealt face down, and two cards are turned face up each turn. The object of the game is to locate pairs of matching cards in as few moves as possible.
```
Comment (dughiboogie) [inline README.md]:
```suggestion
Memory (also known as Concentration) is a card game in which all cards are dealt face down, and two cards are turned face up each turn. The objective of the game is to locate pairs of matching cards in as few moves as possible.
```

Comment (dughiboogie) [reply README.md]: I'm also thinking, usually Memory is played by 2+ players. Should we specify this here?

File: docs/use_cases.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+**Navigation**
+- [Requirements](./requirements.md)
+- [README](../README.md)
+
+**Table of Contents**
+- [UC01 - Application Start](#uc01---application-start)
+  - [Description](#description)
+  - [Generated Requirements](#generated-requirements)
+  - [Implementation Details](#implementation-details)
```
Comment (dughiboogie) [inline docs/use_cases.md]: Should we add Main actor and Pre/Post-conditions to the use cases structure? Or should we wait for when we start writing them more in detail?

**Review by dughiboogie — CHANGES_REQUESTED** [review]
Just a couple small grammar mistakes and some ideas for possible updates.
Looks good!

File: README.md (hunk 1 above)
Comment (GammaSigma1234) [reply README.md]: Yes, I agree. If @DanieleAsnicar also agrees, we can commit the suggestion and close the conversation.

File: docs/use_cases.md (hunk 3 above)
Comment (GammaSigma1234) [reply docs/use_cases.md]: Good catch. Maybe we could discuss this together during the next meeting?

File: README.md (hunk 2 above)
Comment (GammaSigma1234) [reply README.md]: Good remark. I was thinking that, for simplicity's sake, the first prototype might just be a single-player version of the game, but we should definitely start reasoning about the actual number of players as soon as possible. Waiting for @DanieleAsnicar's opinion before closing this conversation. Maybe we should put this on the agenda for the next meeting?

File: README.md (hunk 1 above)
Comment (DanieleAsnicar) [reply README.md]: > If @DanieleAsnicar also agrees, we can commit the suggestion and close the conversation.

 Yes, I agree.

File: README.md (hunk 2 above)
Comment (DanieleAsnicar) [reply README.md]: I referenced it in a new issue

**Review by DanieleAsnicar — APPROVED** [review]

**Review by dughiboogie — APPROVED** [review]

**Pull request merged**